qc-runner/src/dataIntegration.r

```diff
@@ -99,31 +104,36 @@ run_dataIntegration <- function(scdata, config){
     # default for FindIntegrationAnchors
     k.filter <- 200 
 
+    # By defatult, we need RNA assay to compute the integrated matrix.
+    Seurat::DefaultAssay(scdata) <- "RNA"
+
+    # @misc slots not preserved so transfer
+    misc <- scdata@misc
+
     # temporary to make sure we don't run integration if unisample
     nsamples <- length(unique(scdata$samples))
-    if(nsamples>1 && method!="noIntegration"){
+    if(nsamples>1 && method!="unisample"){
         if(method=="seuratv4"){
             # Currently, we only support Seurat V4 pipeline for the multisample integration
-            #FIX FOR CURRENT DATASET!!!!!!
-            Seurat::DefaultAssay(scdata) <- "RNA"
```

Comment:
Agree that this shouldn't be necessary anymore

Reply:
Mmm I think that is good to have, in order to ensure the active assay to `RNA`, but I removed from here and put it before the if statement (to cover also FastMNN)